src/merkle/path.rs

```diff
@@ -82,3 +82,24 @@ impl IntoIterator for MerklePath {
         self.nodes.into_iter()
     }
 }
+
+// MERKLE PATH CONTAINERS
+// ================================================================================================
+
+/// A container for a [Word] value and its [MerklePath] opening.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct ValuePath {
+    /// The node value opening for `path`.
+    pub value: Word,
+    /// The path from `value` to `root` (exclusive).
+    pub path: MerklePath,
+}
+
+/// A container for a [MerklePath] and its [Word] root.
```

Comment:
nit: I think we should add a note here that the root can be computed by verifying the Merkle path against some value, but this structure does not provide any guarantees that this is actually correct.

Reply:
Added!